bt_ddos_shield/miner_shield.py

```diff
@@ -496,54 +498,58 @@ def run(self, miner_shield: MinerShield):
         miner_shield._handle_publish_manifest()
 
 
+class ShieldSettings(BaseSettings):
+    aws_access_key_id: str = Field(min_length=1)
+    aws_secret_access_key: str = Field(min_length=1)
+    aws_region_name: str = Field(min_length=1)
+    """AWS region name where shield will be created"""
+    aws_s3_bucket_name: str = Field(min_length=1)
+    """AWS S3 bucket name where manifest file will be stored"""
+    aws_route53_hosted_zone_id: str = Field(min_length=1)
+    """AWS Route53 hosted zone ID for creating DNS entries for server clients"""
+    aws_miner_instance_id: str = ''
+    """AWS instance ID of miner server"""
+    aws_miner_instance_ip: str = ''
+    """AWS instance IP of miner server"""
+    miner_instance_port: int
+    """Port on which miner server is listening"""
+    sql_alchemy_db_url: str = Field('sqlite:///ddos_shield.db', min_length=1)
+    """SQL Alchemy URL to database where shield state is stored"""
+    miner_hotkey: str = Field(min_length=1)
+    """Hotkey of shielded miner"""
+    options: MinerShieldOptions = MinerShieldOptions()
```

Comment:
how does one specify it using env vars?

Reply:
if there is an option to specifcy nested models then it's all good. Especially if it's mostly for tests